hikari/api/rest.py

```diff
@@ -1154,27 +1458,215 @@ async def execute_webhook(
+            being empty with no attachments or embeds; messages with more than
+            2000 characters in them, embeds that exceed one of the many embed
+            limits; too many attachments; attachments that are too large;
+            invalid image URLs in embeds; users in `user_mentions` not being
+            mentioned in the message content; roles in `role_mentions` not
+            being mentioned in the message content.
+        hikari.errors.UnauthorizedError
+            If you are unauthorized to make the request (invalid/missing token).
+        hikari.errors.NotFoundError
+            If the channel is not found.
+        hikari.errors.InternalServerError
+            If an internal error occurs on Discord while handling the request.
+        """  # noqa: E501 - Line too long
 
     @abc.abstractmethod
     async def fetch_gateway_url(self) -> str:
-        ...
+        """Fetch the gateway url.
+
+        !!! note
+            This endpoint does not require any valid authorization.
+
+        Raises
+        ------
+        hikari.errors.InternalServerError
+            If an internal error occurs on Discord while handling the request.
+        """
 
     @abc.abstractmethod
     async def fetch_gateway_bot(self) -> sessions.GatewayBot:
-        ...
+        """Fetch the gateway sharding requirements for the bot user.
+
+        !!! note
+            The bot user is determined by the token used in the request.
+            Providing a non-`Bot` token will result in undefined behaviour.
+
+        Returns
+        -------
+        hikari.sessions.GatewayBot
+            The gateway bot.
+
+        Raises
+        ------
+        hikari.errors.UnauthorizedError
+            If you are unauthorized to make the request (invalid/missing token).
+        hikari.errors.InternalServerError
+            If an internal error occurs on Discord while handling the request.
+        """
 
     @abc.abstractmethod
     async def fetch_invite(self, invite: invites.Inviteish) -> invites.Invite:
-        ...
+        """Fetch an existing invite.
+
+        Parameters
+        ----------
+        invite : hikari.invites.Inviteish
+            The invite to fetch.
+
+        Returns
+        -------
+        hikari.invites.Invite
+            The requested invite.
+
+        Raises
+        ------
+        hikari.errors.UnauthorizedError
+            If you are unauthorized to make the request (invalid/missing token).
+        hikari.errors.NotFoundError
+            If the invite is not found.
+        hikari.errors.InternalServerError
+            If an internal error occurs on Discord while handling the request.
+        """
 
     @abc.abstractmethod
     async def delete_invite(self, invite: invites.Inviteish) -> None:
-        ...
+        """Delete an existing invite.
+
+        Parameters
+        ----------
+        invite : hikari.invites.Inviteish
+            The invite to delete.
```

Comment:
could we not be more explicit in the description that this may be an object or code so they don't have to go to the invites module docs to work this out?